BeerOverflowWindowsApp/BarProviders/GetBarListFacebook.cs

```diff
@@ -21,52 +21,67 @@ class GetBarListFacebook : IBeerable
         public List<BarData> GetBarsAround(string latitude, string longitude, string radius)
         {
             RegexTools.LocationDataTextIsCorrect(latitude, longitude, radius);
-            var result = GetBarData(latitude, longitude, radius);
-            var barList = FacebookDataToBars(result);
+            var placeList = GetBarData(latitude, longitude, radius);
+            var barList = PlaceListToBarList(placeList);
             return barList;
         }
 
-        private PlacesResponse GetBarData(string latitude, string longitude, string radius)
+        private IEnumerable<Place> GetBarData(string latitude, string longitude, string radius)
         {
-            PlacesResponse result;
-            try
-            {
-                var webClient = new WebClient();
-                var response = webClient.DownloadString(
-                    string.Format(_apiLink, _accessToken, latitude+","+longitude, radius, _requestedFields, _category));
-                result = JsonConvert.DeserializeObject<PlacesResponse>(response);
-            }
-            catch (Exception exception)
-            {
-                throw exception;
-            }
-            return result;
+            var link = string.Format(_apiLink, _accessToken, latitude, longitude, radius, _requestedFields, _category);
+            var jsonStream = GetJsonStream(link);
+            var barList = JsonConvert.DeserializeObject<PlacesResponse>(jsonStream).data;
+            return barList;
+        }
+
+        public async Task<List<BarData>> GetBarsAroundAsync(string latitude, string longitude, string radius)
+        {
+            RegexTools.LocationDataTextIsCorrect(latitude, longitude, radius);
+            var placeList = await GetBarDataAsync(latitude, longitude, radius);
+            var barList = PlaceListToBarList(placeList);
+            return barList;
         }
 
-        private List<BarData> FacebookDataToBars(PlacesResponse resultData)
+        private async Task<List<Place>> GetBarDataAsync(string latitude, string longitude, string radius)
+        {
+            RegexTools.LocationDataTextIsCorrect(latitude, longitude, radius);
+            var link = string.Format(_apiLink, _accessToken, latitude, longitude, radius, _requestedFields, _category);
+            var jsonStream = await GetJsonStreamAsync(link);
+            var barList = JsonConvert.DeserializeObject<PlacesResponse>(jsonStream).data;
+            return barList;
+        }
+
+        private List<BarData> PlaceListToBarList(IEnumerable<Place> resultData)
         {
             var barList = new List<BarData>();
             
```

Comment:
Saw this earlier, that sometimes we leave an empty line before foreach, if, while and other similar statements, and sometimes we don't. I think we could agree on some kind of a standard here, how it should be for real? I personally leave no space.

Reply:
Our standards are so messed up, we really need to talk about them :(